crates/storage/src/lib.rs

```diff
@@ -1,14 +1,91 @@
-pub fn add(left: u64, right: u64) -> u64 {
-    left + right
+use anyhow::Result;
+use parking_lot::RwLock;
+use redb::{ 
+    Builder,
+    Database,
+    Durability,
+    TableDefinition
+};
+mod config;
+
+
+pub(crate) struct ReamDB {
+    db: RwLock<Database>,
```

Comment:
We don't want to wrap a RwLock or Mutex around the database, maybe we should eventually have a queued write system but `RwLock` should probably not be here